_posts/2017-02-02-Rust-1.15.md

```diff
@@ -0,0 +1,273 @@
+deserialized = Pet { name: "Ferris" }
+```
+
+Another common use-case is [Diesel](http://diesel.rs/). Say we wanted to store
+our `Pet`s in a database. We could do that like this:
+
+```rust
+// some extern crate and use lines elided here
+
+#[derive(Queryable)]
+strut Pet {
+    name: String,
+}
+
+fn main() {
+    use diesel_demo::schema::pets::dsl::*;
+
+    let connection = establish_connection();
+    let results = pets 
+        .limit(5)
+        .load::<Pet>(&connection)
+        .expect("Error loading pets");
+
+    println!("Displaying {} pets", results.len());
+    for pet in results {
+        println!("{}", pet.name);
+    }
+}
+```
+
+For full instructions, see [the website][Diesel].
+
+These kinds of libraries are extremely powerful, but rely on custom derive for
+ergonomics. While these libraries _worked_ on Rust stable previously, they were
+not as nice to use, so much so that we often heard from users "I only use
+nightly because of Serde and Diesel." The use of custom derive is one of the
+most widely used nightly-only features. As such, [RFC 1681] was opened in July
+of last year to support this use-case. The RFC was merged in August, underwent
+a lot of development and testing, and now reaches stable today!
+
+[RFC 1681]: https://github.com/rust-lang/rfcs/pull/1681
+
+To find out how to write your own custom derives, see [the chapter of "The Rust
+Programming Language"](https://doc.rust-lang.org/book/procedural-macros.html).
+
+While we've said "Serde and Diesel" a number of times here, there's a lot of
+other cool things you can do with custom derive: see
+[`derive-new`](https://crates.io/crates/derive-new) for another example. Custom
+derive was also known as "macros 1.1", as it includes the infrastructure for
+supporting even more compile-time powers of Rust, nicknamed "macros 2.0."
+Expect to hear more about this space in future releases.
+
+As a smaller improvement, [`?Sized` can now be used in `where`
+clauses](https://github.com/rust-lang/rust/pull/37791). In other words:
+
+```rust
+struct Foo<T: ?Sized> {
+    f: T,
+}
+
+struct Foo<T> where T: ?Sized {
+    f: T,
+}
+```
+
+This second form is now accepted, and is equivalent to the first.
+
+The build system for Rust [has been re-written in Rust, using
+Cargo][rustbuild]. It is now the default. This process has been long, but has
+finally borne fruit. Given that all Rust development happens on the master
+branch, we've been using it since December of last year, and it's working well.
+There is an open PR [to remove the Makefiles entirely][rustbuild-only], landing
+in Rust 1.17. This will pave the way for `rustc` to use packages from
+`crates.io` in the compiler like any other Rust project, and is a further
+demonstration of the maturity of Cargo.
+
+[rustbuild]: https://github.com/rust-lang/rust/pull/37817
+[rustbuild-only]: https://github.com/rust-lang/rust/pull/39431
+
+Rust has gained [Tier 3 support][tiers] for `i686-unknown-openbsd`, `MSP430`,
+and `ARMv5TEr`.
```

Comment:
"ARMv5TEr" -> "ARMv5TE" afaict. These three platforms are not marked up for links, but their links are below.